### DISCUSSION TITLE
Accessing another box

### DISCUSSION THREAD

**Discussion by elisandroesp** [OFFLINE-GmbH/oc-boxes-support#122] [Help] [answered]
How can we access another box data?
I couldn't find a way on the docs.

My use case: the actual `page name` might be updated by a overriden by a `pageTitle` box. 

How can I fetch this `pagetitle->newTitle` from any other box?

**Comment by tobias-kuendig** [comment, marked as answer]
You can access the current boxes page using the `boxesPage` variable in Twig. If you are using PHP code, you can find the `boxesPage` variable set on the controller (i believe the property is named `vars`). 

Once you've got the page, you can access all boxes using the `->boxes` relation, and check if your `pageTitle` box is in there. If so, change the page's title.

Be aware that this logic must be executed somewhere before the page title is rendered in the document, otherwise your change won't have any effect.

Edit: inside the box partial itself, you can access the page using `box.page`

**Comment by elisandroesp** [comment]
Thanks, it worked!
For those struggling with Twig accessing that list of components, this worked for me:
```
{% set pageTitleBox = boxesPage.boxes | filter(v => v.partial == 'pageTitle') | first %}

{# this was specific to my app, but might help #}
{% set pageTitle = pageTitleBox and pageTitleBox.title_overload ? pageTitleBox.title_overload : boxesPage.name %}
```
